src/main/java/org/prebid/server/bidder/mobilefuse/MobilefuseBidder.java

```diff
@@ -0,0 +1,162 @@
+package org.prebid.server.bidder.mobilefuse;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpMethod;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.mobilefuse.ExtImpMobilefuse;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * MobilefuseBidder {@link Bidder} implementation.
+ */
+public class MobilefuseBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpMobilefuse>> MOBILEFUSE_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpMobilefuse>>() {
+            };
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public MobilefuseBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> imps = new ArrayList<>();
+
+        ExtImpMobilefuse firstExtImpMobilefuse = null;
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpMobilefuse extImpMobilefuse = parseImpExt(imp);
+                firstExtImpMobilefuse = firstExtImpMobilefuse == null ? extImpMobilefuse : firstExtImpMobilefuse;
```

Comment:
it will not be the frist one. It will be last valid.
Just create sepparate method for this

Reply:
In Go version theri is method getFirstMobileFuseExtension and there they parse and check if present any error and then use break to quit from loop